integrations/llama_cpp/pyproject.toml

```diff
@@ -28,6 +28,14 @@ classifiers = [
 ]
 dependencies = ["haystack-ai>=2.16.1", "llama-cpp-python>=0.2.87"]
 
+# On macOS GitHub runners, we use a custom index to download pre-built wheels.
+# Installing from source might fail due to missing dependencies (CMake fails with "OpenMP not found")
+[tool.uv]
+index-strategy = "unsafe-best-match"  # this ensures that packages are not only searched in the below index but also in PyPI
+[[tool.uv.index]]
+name = "llama-cpp-python-macos"
+url = "https://abetlen.github.io/llama-cpp-python/whl/metal/"
```

Comment:
Just for my own knowledge could you explain how this roughly works? Does this only get applied when installing this integration on a mac?

Reply:
This only happens on the CI, where the installer is uv. AFAIK, there is not an easy way to do this in Hatch.

In short, defining an index + "unsafe-best-match" makes sure the package is searched on the provided index + PyPI. Then the most appropriate version is installed.

This means that for macOS, the pre-built metal wheels will be installed. For Linux and Windows, the packages available on PyPI will be used.

More info: https://docs.astral.sh/uv/concepts/indexes/

Reply:
Thanks!